### PR TITLE
Support PL 1.5.0

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]
Healthy cleaning 👍

File: flash/core/data/data_pipeline.py [hunk 1]
```diff
@@ -315,16 +331,34 @@ def _get_dataloader(model: "Task", loader_name: str) -> Tuple[DataLoader, str]:
...
+        elif (
+            model.trainer
+            and hasattr(model.trainer, "datamodule")
+            and model.trainer.datamodule
+            and is_overridden(loader_name, model.trainer.datamodule, flash.DataModule)
+        ):
+            dataloader = getattr(model.trainer.datamodule, loader_name, None)
             attr_name = f"trainer.datamodule.{loader_name}"
 
+        elif _PL_GREATER_EQUAL_1_5_0 and model.trainer:
```
Comment (justusschock) [inline flash/core/data/data_pipeline.py:343]:
```suggestion
        elif _PL_GREATER_EQUAL_1_5_0 and model.trainer is not None:
```

more explicit :)

File: flash/core/trainer.py [hunk 2]
```diff
@@ -277,14 +279,31 @@ def request_dataloader(
...
+            if _PL_GREATER_EQUAL_1_5_0:
+                source = getattr(self._data_connector, f"_{stage.dataloader_prefix}_dataloader_source")
+                if (
+                    not source.is_module()
+                    or not isinstance(source.instance, DataModule)
+                    or isinstance(source.instance, (LightningModule, NewDataModule))
+                ):
+                    dataloader = source.dataloader()
+
+            if dataloader is None:
```
Comment (justusschock) [inline flash/core/trainer.py:300]: why only do this if the dataloader is None?

File: flash/core/data/data_pipeline.py [hunk 3]
```diff
@@ -18,24 +18,40 @@
...
 from flash.core.utilities.stages import _RUNNING_STAGE_MAPPING, RunningStage
 
+if not _PL_GREATER_EQUAL_1_5_0:
+    from pytorch_lightning.trainer.connectors.data_connector import _PatchDataLoader
+
 if TYPE_CHECKING:
     from flash.core.model import Task
 
 
+class DataLoaderGetter:
```
Comment (justusschock) [inline flash/core/data/data_pipeline.py:42]: This still wraps and patches the dataloader, right? Is there no way around this?
Will this patch be assigned to back to the loader or just used internally?

File: flash/image/classification/adapters.py [hunk 4]
```diff
@@ -268,7 +268,11 @@ def _convert_dataset(
             devices = 1
             if isinstance(trainer.training_type_plugin, DataParallelPlugin):
                 # when using DP, we need to sample n tasks, so it can splitted across multiple devices.
-                devices = trainer.accelerator_connector.devices
+                if hasattr(trainer, "accelerator_connector"):
+                    accelerator_connector = trainer.accelerator_connector
+                else:
+                    accelerator_connector = trainer._accelerator_connector
```
Comment (justusschock) [inline flash/image/classification/adapters.py:274]:
```suggestion
                if hasattr(trainer, "_accelerator_connector"):
                    accelerator_connector = trainer._accelerator_connector
                else:
                    accelerator_connector = trainer.accelerator_connector
```

Try using the new API first and just fall back to legacy if necessary :)

**Review by justusschock — APPROVED** [review]

File: flash/core/trainer.py (hunk 2 above)
Comment (tchaton) [reply flash/core/trainer.py:300]: This would request the dataloader if it hasn't been loaded yet. This would happen in custom loops I believe.

**Review by tchaton — APPROVED** [review]
dawn ... This is quite painful .

File: flash/image/classification/adapters.py (hunk 4 above)
Comment (ethanwharris) [reply flash/image/classification/adapters.py:283]: Done :smiley:

File: flash/core/data/data_pipeline.py (hunk 3 above)
Comment (ethanwharris) [reply flash/core/data/data_pipeline.py:42]: Yeah, unfortunately this was the only solution for now, but the upcoming data pipeline refactor should remove this patching entirely

File: flash/core/trainer.py (hunk 2 above)
Comment (ethanwharris) [reply flash/core/trainer.py:300]: Refactored it to make it clearer :smiley:

**Pull request merged**